hangman.py

```diff
@@ -0,0 +1,427 @@
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+    
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+    print("Loading word list from file...")
+    # inFile: file
+    inFile = open(WORDLIST_FILENAME, 'r')
+    # line: string
+    line = inFile.readline()
+    # wordlist: list of strings
+    wordlist = line.split()
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+
+def choose_word(wordlist):
+    """
+    wordlist (list): list of words (strings)
+    
+    Returns a word from wordlist at random
+    """
+    return random.choice(wordlist)
+
+# end of helper code
+
+# -----------------------------------
+
+# Load the list of words into the variable wordlist
+# so that it can be accessed from anywhere in the program
+wordlist = load_words()
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    letters_guessed = set(letters_guessed)
+    new_set = set()
+    for symbol in secret_word:
+        new_set.add(symbol)
+    if letters_guessed | new_set == letters_guessed:
+        return True
+    else:
+        return False
+
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    '''
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    new_list = []
+    for symbol in secret_word:
+        if symbol in letters_guessed:
+            new_list.append(symbol)
+        else:
+            new_list.append('_ ')
+    str = ''.join(new_list)
+    return str
+
+
+
+def get_available_letters(letters_guessed):
+    '''
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    al = str(string.ascii_lowercase)
+    new_list = []
+    for symbol in al:
+        new_list.append(symbol)
+
+    for symbol in letters_guessed:
+        new_list.remove(symbol)
+    str_not_used_letters = ''.join(new_list)
+    return str_not_used_letters
+
+
+def score(secret_word, letters_guessed):
+    letters_guessed = set(letters_guessed)
+    new_set = set()
+    for symbol in secret_word:
+        new_set.add(symbol)
+    length = len(new_set)
+    return length
```

Comment:
As I can remember - the score is not only a length

Reply:
done